### PR TITLE
Isaac code editor iframe wrapper component

### PR DESCRIPTION
**Pull request by chrisjpurdy**
This adds the element that opens an `iframe` to [editor.isaaccode.org](https://editor.isaaccode.org), passing it data specified in content.

Adding this will mean that the code editor can be used from the live site (the editor changes haven't been merged yet, so the only way to make or edit this content block type is by running the editor locally on the `feature/interactive-code-blocks` branch)

### CODE REVIEW

**Review by Morpheu5 — CHANGES_REQUESTED** [review]
Looks OK other maybe than the two comments.

File: src/app/services/questions.ts [hunk 1]
```diff
@@ -109,7 +109,7 @@ export function sanitiseInequalityState(state: any) {
         saneState.result.uniqueSymbols = saneState.result.uniqueSymbols.split('').map((l: string) => REVERSE_GREEK_LETTERS_MAP[l] || l).join('');
     }
     if (saneState.symbols) {
-        for (let symbol of saneState.symbols) {
+        for (const symbol of saneState.symbols) {
```
[the remark this answers is not in the document]
Comment (chrisjpurdy) [reply src/app/services/questions.ts:112]: This was automatically done by WebStorm, and didn't disagree with it either so left them in 😅

File: src/app/components/content/IsaacInteractiveCodeSnippet.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                }
+                break;
+            default:
+                if (segueEnvironment !== "PROD") {
+                    console.error("IsaacInteractiveCodeSnippet received a malformed message from the editor iframe!");
+                }
+        }
+    }, [receivedData, segueEnvironment]);
+
+    return <iframe title={"Code Sandbox"} src={"http://localhost:3000/#" + uid.current} ref={iframeRef} onLoad={sendQuestion} className={"isaac-code-iframe w-100 mb-1"} style={
```
[the remark this answers is not in the document]
Comment (chrisjpurdy) [reply src/app/components/content/IsaacInteractiveCodeSnippet.tsx:91]: Definitely not! I'll fix that

File: src/app/services/questions.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Morpheu5) [reply src/app/services/questions.ts:112]: Oh, alright then 😄

**Review by Morpheu5 — APPROVED** [review]

**Pull request merged**